### PR TITLE
Fixed empty query, changed datetime arg

### PR DESCRIPTION
**Pull request by ishanip24**
## Summary:
In cases where there are no blocked requests, the query will return a table with all column values as '(None)'. This PR mainly handles this edge case and will exit the program instead of throwing a type error as it did previously. It also includes documentation and variable naming fixes.

Issue: INFRA-6453

## Test plan:

- Find a date and time where there is a query returning 'None' and ensure that no error comes and that nothing shows up on the Slack bot-testing channel. An example is `python waf_alert.py --datetime '2021-07-23 10:03:00'`

- Find a date and time where there is a query that returns a valid output (i.e. '2021-07-28 16:30:00') and ensure that an alert message comes on the bot-testing Slack channel. An example is `python waf_alert.py --datetime '2021-07-28 16:30:00'`

### CODE REVIEW

File: gae_dashboard/waf_alert.py [hunk 1]
```diff
@@ -135,15 +133,15 @@ def waf_detect(end):
...
 def main():
     # For demoing alert against specific attack time
-    # python waf_alert.py --date '2021-07-23 10:03:00'
+    # python waf_alert.py --datetime '2021-07-23 10:03:00'
     # NOTE: Dates will only work for after 2021-07-20
-    parser = argparse.ArgumentParser(description='Process the date that we want to investigate.')
-    parser.add_argument("--date", help="Date that we want to investigate.",
+    parser = argparse.ArgumentParser(description='Process the date and time that we want to investigate.')
+    parser.add_argument("--datetime", help="Date and time that we want to investigate.",
      type=lambda s: datetime.datetime.strptime(s, TS_FORMAT))
```
Comment (latteier) [inline gae_dashboard/waf_alert.py:140]: nit: might be worth mentioning in the help for the argument the required format

**Review by latteier — APPROVED** [review]
Looks good.

I see a couple things that could be cleaned up, but otherwise looks good.

**Review by LoveAndCoding — APPROVED** [review]
Few small comments, but looks good 🥍  (random emoji since I don't have a random Pokemon macro here)

File: gae_dashboard/waf_alert.py [hunk 2]
```diff
@@ -117,15 +114,16 @@ def waf_detect(end):
     message_percentage = results[0]['message_percentage']
     rua_percentage = results[0]['rua_percentage']
 
+    if ip_percentage == '(None)':
```
[the remark this answers is not in the document]
Comment (dat-boris) [reply gae_dashboard/waf_alert.py:117]: I agree that some explanation here is useful.  It seems like it is odd to get a row of data with string `(None)` when you are querying the numerical field - especially we already filter for `not results` above?

File: gae_dashboard/waf_alert.py [hunk 3]
```diff
@@ -89,14 +89,11 @@
...
-    try:
-        query = QUERY_TEMPLATE.format(
-            fastly_log_tables=log_name,fastly_waf_log_tables=waf_log_name,
-            start_timestamp=start.strftime(TS_FORMAT),
-            end_timestamp=end.strftime(TS_FORMAT),
-            )
-    except TypeError:
-        query=None
+    query = QUERY_TEMPLATE.format(
+        fastly_log_tables=log_name,fastly_waf_log_tables=waf_log_name,
```
Comment (dat-boris) [inline gae_dashboard/waf_alert.py:93]: I think you already catch this bug, but these table names still refer to today's timestamp.

As mentioned in https://github.com/Khan/internal-webserver/pull/7#discussion_r671468582 - I also have concerns about this crossing day boundary.  But let's work on it with #10 instead.

nit: The formatting here we should break after the comma into new line.  It is unfortunate we don't have linter on this repo.

**Review by dat-boris — COMMENTED** [review]
Looks good - thank you for the great work!  I think you already mentioned one of the bug you caught.  I would like to see that commit before approving.

File: gae_dashboard/waf_alert.py [hunk 4]
```diff
@@ -27,17 +28,28 @@
...
+parser = argparse.ArgumentParser(description='Process the date and time that we want to investigate.')
+parser.add_argument("--endtime", help="Date and time that we want to investigate. For example, `python waf_alert.py --endtime '2021-07-23 10:03:00'``",
+ type=lambda s: datetime.datetime.strptime(s, TS_FORMAT))
+args = parser.parse_args()
+
+if args.endtime:
+    now = args.endtime
+else:
+    now = datetime.datetime.utcnow()
+ymd = now.strftime(TABLE_FORMAT)
```
Comment (dat-boris) [inline gae_dashboard/waf_alert.py:44]: In general, we would prefer to minimize doing argument parsing in global scope.  This makes the code less clear and less reusable.

File: gae_dashboard/waf_alert.py [hunk 5]
```diff
@@ -64,15 +76,16 @@
...
   GROUP BY
     t
   ORDER BY
     t DESC
 )
 SELECT
-  SUM(blocked) AS blocked,
-  SUM(total) AS total,
-  100 * SUM(blocked) / SUM(total) AS percentage,
+  100*MAX(blocked/total) AS percentage,
```
Comment (dat-boris) [inline gae_dashboard/waf_alert.py:88]: See my comment at https://github.com/Khan/internal-webserver/pull/7/files#r671471633 - i think this still suffers the same issue.

**Review by dat-boris — COMMENTED** [review]
I still see a few issues, but let's work on that in #10 😄
